### ISSUE TITLE
"illegal hardware instruction" error

### ISSUE THREAD

**Issue by ori-scala** [bioconda/bioconda-recipes#52034]
After installing the latest MUSCLE version (5.2), I'm getting the following error when trying to run MUSCLE.

`illegal hardware instruction`

I assume it's related to the Apple M series chipset, I'm using Apple M1 Pro.

To reproduce:

`conda install bioconda::muscle==5.2.0 -y`

Then run `muscle --help` and get:

```
muscle --help
[1]    14859 illegal hardware instruction  muscle --help
```

Then, downgrade using:

`conda install bioconda::muscle==5.1.0 -y`

and `muscle --help` works again with the following output:

```
muscle --help

muscle 5.1.osx64 []  34.4Gb RAM, 10 cores
Built May 15 2023 21:29:02
(C) Copyright 2004-2021 Robert C. Edgar.
https://drive5.com

This is help.txt
```

**Comment by apcamargo** [comment]
Does `-super5` work? [It is included in the tests](https://github.com/bioconda/bioconda-recipes/blob/44837e97898c82ba3279254014c0739b22c40736/recipes/muscle/run_test.sh).

**Comment by ori-scala** [comment]
Nope, same error
```
[1]    42600 illegal hardware instruction  muscle -super5  -output  -log
```

**Comment by apcamargo** [comment]
Ok, so the problem might be on your end. Is your Mac a M series? Because the output of your `muscle --help` shows `osx64`, which is Intel. Maybe the binary of the wrong architecture is being installed.

How are you installing `muscle`?

**Comment by ori-scala** [comment]
The mac indeed has an M series chipset (M1 Pro) but I'm setting the conda env to osx64 using `conda config --env --set subdir osx-64` as other packages I'm using aren't built for M series and I'm forced to do that.

**Comment by apcamargo** [comment]
I guess the other binaries are being executed through Rosetta? If so, I don't know why `muscle` isn't (assuming the issue is that the issue is that the binary is not compatible with the processor's architecture).

I don't think I know enough about this to help you. If you're using Pixi, you can try to ask for help in their Discord (people there are very helpful). Otherwise, maybe @bioconda/osx can help?

**Comment by ori-scala** [comment]
You're correct, the other binaries are being executed through Rosetta. It doesn't happen with other packages nor with any muscle version prior to 5.2

**Comment by apcamargo** [comment]
In this case, I don't know what is going on. It could be something with the macOS ARM CI, but the core team would know better than me.

**Comment by ori-scala** [comment]
Ok, thank you anyway Antônio!

**Comment by apcamargo** [comment]
If you create an environment only for muscle and install the ARM version, does it work?

**Comment by ori-scala** [comment]
Yep, that works as expected.

**Comment by apcamargo** [comment]
This explains why the tests didn't catch this. From my understanding, the only major difference between the Bioconda build and Robert's build is that he is using GCC and Bioconda uses clang. As far as I know, there's no way for us to use GCC to build for macOS, but I don't really know our building system that well.

As you said, version 5.1 worked in Rosetta and it seems he used Clang back then (I'm inferring this from his CI). But the code changed a lot since then...

**Comment by apcamargo** [comment]
Can you update the issue title to something that reflects better the info we gathered?

**Comment by ori-scala** [comment]
@apcamargo updated the title as requested

**Comment by milot-mirdita** [comment]
Seems muscle is using AVX(2/512/...) in version 5.2. Rosetta does not emulate AVX.
Here is likely the issue:
https://github.com/bioconda/bioconda-recipes/blob/b310cef61eebed800970d596e79e059c8d9b91d5/recipes/muscle/vcxproj_make.py#L74C1-L75C43

Its probably not a good idea to use march native for conda packages.

For MMseqs2 I was explicitly setting to use up to SSE4_1 in comda, with the latest release 16 I made MMseqs2 use AVX2 in conda, so the same issue would likely happen now if someone were to use it in a Rosetta emulated environment.

**Comment by apcamargo** [comment]
Thanks, @milot-mirdita! I had no idea Rosetta doesn't emulate AVX.

I guess this is. Matter of how much MUSCLE benefits from AVX. If the performance increase is substantial, it might be worth it to leave as is. Perhaps @rcedgar has a thought about this

**Comment by milot-mirdita** [comment]
No matter what, conda recipes shouldn't be compiled with march native, since this will result in binaries that use the instruction set of whatever GitHub runner happens to compile the binary. Most likely recent intel machines with avx512.

Conda's default is I think nocona, which is ancient. A good compromise might be using the new x86-64-v3 psABI specifier.

**Comment by rcedgar** [comment]
I don't think muscle has any explicit SIMD code, if I'm not forgetting something then any AVX instructions are compiler optimizations.
